### PR TITLE
DCS-706 Moving prison api to typescript class 

### PR DESCRIPTION
**Pull request by andrewrlee**
Also creating type definition files for whereabouts and prison modules:

Created prison api types via: 
```npx @manifoldco/swagger-to-ts:  https://api-dev.prison.service.justice.gov.uk/api/swagger.json --output prison-api.ts```

### CODE REVIEW

File: backend/api/prisonApi.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import type { components } from 'prisonApi'
+import { Response } from 'superagent'
+import contextProperties from '../contextProperties'
+import { mapToQueryString } from '../utils'
+import Client from './oauthEnabledClient'
+
+type Context = any
```
Comment (bell-pa) [inline backend/api/prisonApi.ts:7]: Context is used throughout the application - so this is probably not the right place to declare this type.
It should be opaque to most of the code (possibly by using scope to control access?...

Comment (andrewrlee) [reply backend/api/prisonApi.ts:7]: Yes, I'm not quite ready to provide the typings for this as that code is used everywhere - I've started to extract it out so specifying it and moving it to a better place will be easier. Haven't quite decided on the best place yet either.

Comment (andrewrlee) [reply backend/api/prisonApi.ts:7]: I think we barely use it in this project apart from it storing the access token (no pagination)

**Review by bell-pa — APPROVED** [review]
LGTM

File: backend/api/prisonApi.ts (hunk 1 above)
Comment (bell-pa) [reply backend/api/prisonApi.ts:7]: That is what it is for. Passing the token from incoming to outgoing requests. and tunnelling other stuff like that.

**Pull request merged**